lua/autorun/client/spraymesh_manager.lua

```diff
@@ -277,8 +314,69 @@ function SprayMeshManager()
             SprayMeshManagerThumbnails()
         end
     end
+	
+    SprayMeshManagerInput = vgui.Create("DTextEntry", SprayMeshManagerBase)
+    SprayMeshManagerInput:SetSize(245, 16)
+    SprayMeshManagerInput:SetPos(15, 665)
+	
+    SprayMeshManagerCheckbox = vgui.Create("DCheckBoxLabel", SprayMeshManagerBase)
+    SprayMeshManagerCheckbox:SetSize(16, 16)
+    SprayMeshManagerCheckbox:SetPos(430 * .3 - 90, 690)
+    SprayMeshManagerCheckbox:SetText("Is this spray pornograhic?")
+	
+    SprayMeshManagerInputButton = vgui.Create("DButton", SprayMeshManagerBase)
+    SprayMeshManagerInputButton:SetSize(40, 20)
+    SprayMeshManagerInputButton:SetPos(430 * .3 + 70, 688)
+    SprayMeshManagerInputButton:SetText("OK")
+	
+    function SprayMeshManagerInputButton:DoClick()
+		SprayList[#SprayList+1] = {[1] = SprayMeshManagerInput:GetValue(),[2] = (SprayMeshManagerCheckbox:GetChecked() and 1) or 0} --why the hell is unchecked equal to nil?
+		ReloadManager()
+    end
+	
+	SprayMeshManagerPreview = vgui.Create("DHTML", SprayMeshManagerBase)
+    SprayMeshManagerPreview:SetSize(100, 100)
+    SprayMeshManagerPreview:SetPos(290, 615)
+	SprayMeshManagerInput.OnChange = function(self)
+		SanitizeImgurIdAsync(self:GetValue(),function(id)
```

Comment:
Were you able to get ImgurMaterial() to work? You can literally do this in, like, 2 lines...

Reply:
I have no idea what you're talking about